src/app/pages/courses/editCourse/editCourse.component.ts

```diff
@@ -0,0 +1,56 @@
+import {
+  Component,
+  OnInit,
+  Input,
+  Output,
+  NgZone,
+  EventEmitter,
+  ChangeDetectionStrategy
+} from '@angular/core';
+
+import { Observable } from 'rxjs/Observable';
+import { AppState } from '../../../app.service';
+import { ICourse } from '../iCourse.interface';
+
+@Component({
+  selector: 'app-edit',
+  templateUrl: './editCourse.component.html',
+  styleUrls: ['./editCourse.component.html'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+
+export class EditCourseComponent implements OnInit {
+  @Input() courseData: ICourse;
+  @Output() courseDataOutput = new EventEmitter<any>();
+  constructor(
+    private ngZone: NgZone,
+  ) {
+    console.log('test');
+    this.courseData = {
+      id: null,
+      name: null,
+      time: null,
+      date: null,
+      description: null
+    };
+    console.log('this.ngZone 2');
+    console.log(this.ngZone);
+    console.log(this.ngZone.onStable);
```

Comment:
wiem, że opis dotyczący ngZone był beznadziejny w tasku, ale z tego co zrozumiałem to chodziło o wyliczenie czasu miedzy stanem 'unstable' a 'stable' i można to zrobić poprzez odpalenie timestampów dla tych dwóch eventów, np.
```
                // this.ngZone.onUnstable.subscribe(() => {
		// 	this.start = new Date();
		// 	console.log('onUnstable', this.start.getTime());
		// });

		// this.ngZone.onStable.subscribe(() => {
		// 	const d = new Date();
		// 	console.log('onStable', d.getTime());
		// 	if (this.start) {
		// 		console.log('onUnstable - onStable ', d.getTime() - this.start.getTime());
		// 	}
		// });
```